### PR TITLE
Revamp design page

### PR DESCRIPTION
**Pull request by ektravel**
### Description

This PR revamps the Design section of the documentation as follows:

- Adds a new subsection called Services to the sidebar
- Adds an updated Druid architecture diagram
- Moves "Storage design" section out of design.md and into its own topic
- Moves "Query processing" section out of design.md and into its own topic
- Consolidates the information in "Design" and "Processes and servers" topics
- Removes "Processes and servers" topic from the sidebar
- Replaces "process" with "service"
- Adds a redirect for processes.md

### Preview

https://druid-98c2egkpg-katyamacedo-bigcommerce.vercel.app/docs/latest/design/architecture

This PR has:

- [x] been self-reviewed.

### CODE REVIEW

**Review by techdocsmith — CHANGES_REQUESTED** [review]
I think this is mostly OK. However, I think it would be worth it to get clarity on the router, which seems really out of date.

File: docs/design/broker.md [hunk 1]
```diff
@@ -25,18 +26,18 @@ title: "Broker"
...
 ### HTTP endpoints
 
 For a list of API endpoints supported by the Broker, see [Broker API](../api-reference/legacy-metadata-api.md#broker).
 
 ### Overview
 
-The Broker is the process to route queries to if you want to run a distributed cluster. It understands the metadata published to ZooKeeper about what segments exist on what processes and routes queries such that they hit the right processes. This process also merges the result sets from all of the individual processes together.
-On start up, Historical processes announce themselves and the segments they are serving in Zookeeper.
+The Broker is the service to route queries to if you want to run a distributed cluster. It understands the metadata published to ZooKeeper about what segments exist on what services and routes queries such that they hit the right services. This service also merges the result sets from all of the individual services together.
+On start up, Historical services announce themselves and the segments they are serving in ZooKeeper.
```
[the remark this answers is not in the document]
Comment (ektravel) [reply docs/design/broker.md:40]: That's a good question. I didn't add/change that but I'll look into updating the text.

File: docs/design/architecture.md [hunk 2]
```diff
@@ -102,225 +178,4 @@ For more details, please see the [Metadata storage](../design/metadata-storage.m
 
 Used for internal service discovery, coordination, and leader election.
 
-For more details, please see the [ZooKeeper](zookeeper.md) page.
-
-
-## Storage design
```
[the remark this answers is not in the document]
Comment (ektravel) [reply docs/design/architecture.md:108]: Added "Learn more"

File: docs/design/architecture.md [hunk 3]
```diff
@@ -23,45 +23,121 @@ title: "Design"
...
+
+A Query server provides the endpoints that users and client applications interact with, routing queries to Data servers or other Query servers (and optionally proxied Master server requests).
+
+Query servers divide operations between Broker and Router services.
+
+#### Broker service
+
+[Broker](../design/broker.md) services receive queries from external clients and forward those queries to Data servers. When Brokers receive results from those subqueries, they merge those results and return them to the caller. End users typically query Brokers rather than querying Historical or MiddleManager services on Data servers directly.
+
+#### Router service (optional)
```
[the remark this answers is not in the document]
Comment (ektravel) [reply docs/design/architecture.md]: Updated

File: docs/design/broker.md [hunk 4]
```diff
@@ -25,18 +26,18 @@ title: "Broker"
...
 
 ### HTTP endpoints
 
 For a list of API endpoints supported by the Broker, see [Broker API](../api-reference/legacy-metadata-api.md#broker).
 
 ### Overview
 
-The Broker is the process to route queries to if you want to run a distributed cluster. It understands the metadata published to ZooKeeper about what segments exist on what processes and routes queries such that they hit the right processes. This process also merges the result sets from all of the individual processes together.
-On start up, Historical processes announce themselves and the segments they are serving in Zookeeper.
+The Broker is the service to route queries to if you want to run a distributed cluster. It understands the metadata published to ZooKeeper about what segments exist on what services and routes queries such that they hit the right services. This service also merges the result sets from all of the individual services together.
```
[the remark this answers is not in the document]
Comment (ektravel) [reply docs/design/broker.md]: Updated header levels

**Review by techdocsmith — APPROVED** [review]
LGTM

File: website/redirects.js [hunk 5]
```diff
@@ -310,11 +310,14 @@ const Redirects=[
     "from":  "/docs/latest/operations/api-reference.html",
     "to": "/docs/latest/api-reference/"
   },
+  {
+    "from": "/docs/latest/design/processes.html",
```
[the remark this answers is not in the document]
Comment (ektravel) [reply website/redirects.js:314]: Added a redirect and removed [processes.md](https://github.com/ektravel/druid/blob/sidebar-add-services/docs/design/processes.md).

File: docs/design/storage.md [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+  ~ software distributed under the License is distributed on an
+  ~ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  ~ KIND, either express or implied.  See the License for the
+  ~ specific language governing permissions and limitations
+  ~ under the License.
+  -->
+
+This topic describes the components integral to data storage in Apache Druid.
+
+## Datasources and segments
```
[the remark this answers is not in the document]
Comment (ektravel) [reply docs/design/storage.md:27]: Updated.

**Review by vtlim — APPROVED** [review]

**Comment by a2l007** [comment]
Looks like the word Historicals is misspelt on the architecture diagram.
@ektravel If you have the source image handy, It would be great if you could fix this.

**Comment by ektravel** [comment]
@a2l007 Thank you for bringing it to my attention. I will update the source image.

**Pull request merged**